includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -1435,6 +1441,35 @@ private function finalize_styles() {
 		}
 	}
 
+	/**
+	 * Ampify CSS selectors.
+	 *
+	 * @param Selector $selector CSS Selector.
+	 */
+	private function ampify_selector( $selector ) {
```

Comment:
@westonruter Reworked the logic to replace the components before tree shaking instead (compared to the previous commit). Wondering if it's OK to just replace the selectors or is it possible that this way we'll be replacing a selector that's actually in use.

Also, in case of img it can be replaced with either `amp-img` or `amp-playbuzz`, thinking if we should add an additional selector for potential `amp-anim`, too.

Reply:
It would be ideal if the selector mappings could be added to each sanitizer. For example, on `AMP_Img_Sanitizer` there could be a `get_selector_mapping()` method which returns:

```php
array(
    'img' => array( 'amp-img', 'amp-anim' )
)
```

Whereas on the `AMP_Audio_Sanitizer` class it would return:

```php
array(
    'video' => array( 'amp-video ),
)
```

In other words, there could be multiple AMP components that could be converted so the selector should do replacements for each one, potentially duplicating a selector however many times is needed.

As for how to get the `AMP_Style_Sanitizer` to be able to invoke such a `get_selector_mapping` method on the other sanitizers, it could be done similar to as proposed for passing embed handlers to an embed sanitizer: https://github.com/Automattic/amp-wp/pull/1128/files#diff-2585472f207548f626a43a7ff3cab922R150

I have a concern with that approach of passing class instances as sanitizer args, however. It could cause problems with the response caching introduced in #1156, since the cache key is computed by passing the sanitizer args:

https://github.com/Automattic/amp-wp/blob/ab7204af35a8eb68bd8840a2e3531df8ae73c21f/includes/class-amp-theme-support.php#L1073-L1078

/cc @juanchaur1